provider/src/main/java/com/raygun/raygun4android/rum/RUM.java

```diff
@@ -19,225 +19,136 @@
 import com.raygun.raygun4android.network.RaygunNetworkLogger;
 import com.raygun.raygun4android.services.RUMPostService;
 
-import java.lang.ref.WeakReference;
 import java.text.SimpleDateFormat;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.temporal.ChronoUnit;
 import java.util.Calendar;
 import java.util.TimeZone;
 import java.util.UUID;
-import java.util.concurrent.TimeUnit;
 
-public class RUM implements ActivityLifecycleCallbacks {
-    private static RUM rum;
-    private static WeakReference<Activity> mainActivity;
-    private static WeakReference<Activity> currentActivity;
-    private static WeakReference<Activity> loadingActivity;
-    private static long startTime;
-    private static long lastSeenTime;
-    private static String sessionId;
-    private static RaygunUserInfo currentSessionUser;
+public class RUM {
+    // Singleton instance
+    private static RUM instance;
 
-    private static void attach(Activity mainActivity) {
+    private long lastSeenTime;
+    private String sessionId;
+    private RaygunUserInfo currentSessionUser;
+    private final RUMActivity rumActivity;
 
-        RaygunLogger.v("attached RUM");
-        if (RUM.rum == null && mainActivity != null) {
-            Application application = mainActivity.getApplication();
-
-            if (application != null) {
-                RUM.mainActivity = new WeakReference<>(mainActivity);
-                RUM.currentActivity = new WeakReference<>(mainActivity);
-                RUM.startTime = System.nanoTime();
-
-                RUM.rum = new RUM();
-                application.registerActivityLifecycleCallbacks(RUM.rum);
-
-                if (doesNeedSessionRotation()) {
-                    rotateSession(currentSessionUser, currentSessionUser);
-                }
-
-                RaygunNetworkLogger.init();
-            }
-        }
-
-        RUM.lastSeenTime = System.currentTimeMillis();
-    }
-
-    static void attach(Activity mainActivity, boolean networkLogging) {
-        RaygunNetworkLogger.setEnabled(networkLogging);
-        attach(mainActivity);
+    private RUM() {
+        this.rumActivity = new RUMActivity(this, new RUMFragment(this));
     }
 
-    protected static void detach() {
-        if (RUM.rum != null
-            && RUM.mainActivity.get() != null
-            && RUM.mainActivity.get().getApplication() != null) {
-            RUM.mainActivity.get().getApplication().unregisterActivityLifecycleCallbacks(RUM.rum);
-            RUM.mainActivity = null;
-            RUM.currentActivity = null;
-            RUM.rum = null;
+    /**
+     * Gets the singleton instance of the RUM class.
+     */
+    public static RUM getInstance() {
```

Comment:
`RUM` no longer uses static methods, instead access is done with `getInstance()`. This does not affect end users, but internally allows mocking the RUM instance. For example, in `RUMFragmentTest` the `RUM` instance is mocked.